tests/native_api_tests/test_api_meta_data.cpp

```diff
@@ -67,6 +67,7 @@ TEST_CASE("API Meta Data") {
 
             try {
                 MetaData::get_attribute_by_name("No_dataset", "no_name", "no attribute");
+                throw std::exception{};
```

Comment:
These are added across the API tests (where needed) to prevent this bug from happening again.

Reply:
Sonar cloud [complains](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&pullRequest=795&types=CODE_SMELL&inNewCodePeriod=true&sinceLeakPeriod=true&id=PowerGridModel_power-grid-model) about these. @Jerry-Jinfeng-Guo, @mgovers I think it is okay to bypass sonar here, but maybe you have a different suggestion?

Reply:
I think we can simply define a test specific exception to shut it:
```
class APItestException : public std::exception {
public:
    char* const  what() const noexcept override {
        return "Exception occurred in API tests";
    }
};
```

Reply:
I like that better, the only thing that I am not very fond of is that it probably needs to be placed on each test file, but that is just a minor thing. I'll make the fix.

Reply:
custom exception types like `Unreachable` or similar should be used in production, but in test, please go with the functionality that is designed for specifically that purpose

Reply:
This was addressed in [43646b2](https://github.com/PowerGridModel/power-grid-model/pull/795/commits/43646b2fbad8e0a453a1a8990f318a597ebee034). Hopefully it is more robust and clearer now.

Reply:
Awesome!